### PR TITLE
Add logged check in Safari #1027

### LINKED ISSUE #1027: Cookies arn't included in requests
## Checklist

- [ ] I'm reporting a bug unrelated to a specific site
- [ ] I've verified that I'm running yt-dlp version **2021.09.02**
- [ ] I've checked that all provided URLs are alive and playable in a browser
- [ ] The provided URLs do not contain any DRM to the best of my knowledge
- [ ] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [ ] I've searched the bugtracker for similar bug reports including closed ones
- [ ] I've read bugs section in FAQ

## Verbose log

```
[debug] Command-line config: ['--config-location', 'D:\\Utils/YoutubeDL/Configs/ytdlp_oreilly.conf', '--playlist-items', '2']
[debug] Batch file urls: ['https://learning.oreilly.com/videos/the-principles-of/9781491935811/']
WARNING: You have asked for unplayable formats to be listed/downloaded. This is a developer option intended for debugging. If you experience any issues while using this option, DO NOT open a bug report
WARNING: --add-metadata is ignored since --allow-unplayable-formats was given
WARNING: --embed-subs is ignored since --allow-unplayable-formats was given
WARNING: --embed-thumbnail is ignored since --allow-unplayable-formats was given
[Cookies] Extracting cookies from firefox
[debug] Extracting cookies from: "C:\Users\mh-sh\AppData\Roaming\Mozilla\Firefox\Profiles\8tceuger.default-release\cookies.sqlite"
[Cookies] Extracted 1589 cookies from firefox
[debug] Loading archive file 'archive.txt'

[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.09.02 (source)
[debug] Plugin Extractors: ['SamplePlugin']
[debug] Git HEAD: 57aa7b851
[debug] Python version 3.9.1 (CPython 64bit) - Windows-10-10.0.19041-SP0
[debug] exe versions: ffmpeg n4.4-4-gacb339bb88, ffprobe n4.4-4-gacb339bb88
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
[debug] [safari:course] Extracting URL: https://learning.oreilly.com/videos/the-principles-of/9781491935811/
[safari:course] 9781491935811: Downloading course JSON
[download] Downloading playlist: The Principles of Microservices
[info] Writing playlist metadata as JSON to: The Principles of Microservices\0. The Principles of Microservices.info.json
WARNING: There's no playlist description to write.
[safari:course] playlist The Principles of Microservices: Collected 15 videos; downloading 1 of them
[download] Downloading video 1 of 1
[debug] [safari:api] Extracting URL: https://learning.oreilly.com/api/v1/book/9781491935811/chapter/video221406.html
[safari:api] 9781491935811/video221406: Downloading part JSON
[debug] [safari] Extracting URL: https://learning.oreilly.com/videos/the-principles-of/9781491935811/9781491935811-video221406
[debug] [Kaltura] Extracting URL: https://cdnapisec.kaltura.com/html5/html5lib/v2.37.1/mwEmbedFrame.php?wid=_1926081&uiconf_id=29375172&flashvars%5BreferenceId%5D=9781491935811-video221406
[Kaltura] 9781491935811-video221406: Downloading webpage
[Kaltura] 0_1i6jb4o4: Downloading video info JSON
[Kaltura] 0_1i6jb4o4: Checking mp4-4468 URL
[Kaltura] 0_1i6jb4o4: Downloading m3u8 information
WARNING: [Kaltura] Ignoring subtitle tracks found in the HLS manifest; if any subtitle tracks are missing, please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Downloading subtitles: en
[info] Available formats for 0_1i6jb4o4:
ID       EXT RESOLUTION FPS |  FILESIZE    TBR PROTO  | VCODEC    VBR ACODEC     ABR MORE INFO
-------- --- ---------- --- - ---------- ----- ------ - ------- ----- --------- ---- ---------
hls-58   mp4 audio only     |              58k m3u8_n |               mp4a.40.2  58k
mp4-56   mp4 audio only 0   | ~2.16MiB     56k http   |               unknown    56k isom
hls-425  mp4 640x360        |             425k m3u8_n | unknown  425k unknown     0k
hls-571  mp4 640x360        |             571k m3u8_n | unknown  571k unknown     0k
hls-782  mp4 640x360        |             782k m3u8_n | unknown  782k unknown     0k
mp4-385  mp4 640x360    29  | ~14.80MiB   385k http   | avc1     385k unknown     0k isom
mp4-525  mp4 640x360    29  | ~20.20MiB   525k http   | avc1     525k unknown     0k isom
mp4-726  mp4 640x360    29  | ~27.90MiB   726k http   | avc1     726k unknown     0k isom
hls-1319 mp4 1280x720       |            1319k m3u8_n | unknown 1319k unknown     0k
mp4-1240 mp4 1280x720   29  | ~47.60MiB  1240k http   | avc1    1240k unknown     0k isom
mp4-1966 mp4 1280x720   29  | ~75.40MiB  1966k http   | avc1    1966k unknown     0k isom
mp4-3082 mp4 1920x1080  29  | ~118.00MiB 3082k http   | avc1    3082k unknown     0k isom
mp4-4468 mp4 1920x1080  29  | ~171.00MiB 4468k http   | avc1    4468k unknown     0k mp42
[download] Finished downloading playlist: The Principles of Microservices


```

## Description

I tried download oreilly videos using cookies but the downloader is only invoked on the trial duration meaning that the cookies weren't included in the requests and we are treated as not logged in.
The download using user/pass works tho
Relevent info: [here](https://github.com/yt-dlp/yt-dlp/issues/586#issuecomment-920814101)

### PR DESCRIPTION
**Pull request by Bricio**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fix #1027 to use cookies without -u to bypass problem.

### Working evidences

Without cookies. Only download 1 minute
```
C:\yt-dlp>python -m yt_dlp "https://learning.oreilly.com/videos/mastering-jmeter-5-0/9781838640880/" --playlist-items 5
[safari:course] 9781838640880: Downloading course JSON
[download] Downloading playlist: Mastering JMeter 5.0
[safari:course] playlist Mastering JMeter 5.0: Collected 52 videos; downloading 1 of them
[download] Downloading video 1 of 1
[safari:api] 9781838640880/video1_5: Downloading part JSON
[safari] Downloading login page
[Kaltura] 9781838640880-video1_5: Downloading webpage
[Kaltura] 0_2fuux6mp: Downloading video info JSON
[Kaltura] 0_2fuux6mp: Checking mp4-532 URL
[Kaltura] 0_2fuux6mp: Downloading m3u8 information
[info] 0_2fuux6mp: Downloading 1 format(s): mp4-594
[download] Destination: Response Assertion [0_2fuux6mp].mp4
[download] 100% of 2.33MiB in 00:00
[download] Finished downloading playlist: Mastering JMeter 5.0
```

With cookies. Download full video.
```
C:\yt-dlp>python -m yt_dlp --cookies-from-browser chrome "https://learning.oreilly.com/videos/mastering-jmeter-5-0/9781838640880/" --playlist-items 5
[Cookies] Extracting cookies from chrome
[Cookies] Extracted 1102 cookies from chrome
[safari:course] 9781838640880: Downloading course JSON
[download] Downloading playlist: Mastering JMeter 5.0
[safari:course] playlist Mastering JMeter 5.0: Collected 52 videos; downloading 1 of them
[download] Downloading video 1 of 1
[safari:api] 9781838640880/video1_5: Downloading part JSON
[safari] Downloading login page
[safari] 9781838640880-video1_5: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_5: Downloading webpage
[Kaltura] 0_2fuux6mp: Downloading video info JSON
[Kaltura] 0_2fuux6mp: Checking mp4-532 URL
[Kaltura] 0_2fuux6mp: Downloading m3u8 information
[info] 0_2fuux6mp: Downloading 1 format(s): mp4-594
[download] Destination: Response Assertion [0_2fuux6mp].mp4
[download] 100% of 25.71MiB in 00:01
[download] Finished downloading playlist: Mastering JMeter 5.0
```

### CODE REVIEW

**Comment by pukkandan** [comment]
Superseeded by #3499

File: yt_dlp/extractor/safari.py [hunk 1]
```diff
@@ -150,6 +151,10 @@ def _real_extract(self, url):
             'flashvars[referenceId]': reference_id,
         }
 
+        if not self.LOGGED_IN:
+            if self.is_logged():
+                self.LOGGED_IN = True
+
         if self.LOGGED_IN:
```
Comment (sulyi) [inline yt_dlp/extractor/safari.py:158]: Needless. It's enough to set `self.LOGGED_IN` it in `_perform_login`.

Comment (Bricio) [reply yt_dlp/extractor/safari.py:158]: Not true. That is the hole point of this change and the #1027
If you only set in perform_login cookies are not used if you dont use username + password.

This PR is to check, using cookies, if the user is logged in and to use cookies properly. That is the only option for SSO users that cannot use username + password login.

Comment (sulyi) [reply yt_dlp/extractor/safari.py:158]: Understood, and I completely agree: df6e7e70f2469409f877943ff03cfabf03667a11.
It gets set in `_initialize_pre_login`.

Comment (Bricio) [reply yt_dlp/extractor/safari.py:158]: Is `_initialize_pre_login` called if user only uses cookies?

Try download full videos ( not only first minute ) using your code and using cookies. If it works than this code is already merged in your PR. This PR and the #1027 will be closed as well.

Comment (sulyi) [reply yt_dlp/extractor/safari.py:158]: I believe that's correct. Could you check?
It downloads a ~6:00 long video.
[create-cookie-output.txt](https://github.com/yt-dlp/yt-dlp/files/8603303/create-cookie-output.txt)
[output.txt](https://github.com/yt-dlp/yt-dlp/files/8603304/output.txt)

Comment (Bricio) [reply yt_dlp/extractor/safari.py:158]: Yea, it is working. GJ

```
C:\yt-dlp-3dcac0e131bf3d58fae9c2362c1c526594ff2dcf>python -m yt_dlp --cookies-from-browser chrome "https://learning.oreilly.com/videos/mastering-jmeter-5-0/9781838640880/"
[Cookies] Extracting cookies from chrome
[Cookies] Extracted 1143 cookies from chrome
[safari:course] Checking if logged in
[safari:course] 9781838640880: Downloading course JSON
[download] Downloading playlist: Mastering JMeter 5.0
[safari:course] playlist Mastering JMeter 5.0: Collected 52 videos; downloading 52 of them
[download] Downloading video 1 of 52
[safari:api] Checking if logged in
[safari:api] 9781838640880/video1_1: Downloading part JSON
[safari] Checking if logged in
[safari] 9781838640880-video1_1: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_1: Downloading webpage
[Kaltura] 0_j24uni29: Downloading video info JSON
[Kaltura] 0_j24uni29: Checking mp4-521 URL
[Kaltura] 0_j24uni29: Downloading m3u8 information
[info] 0_j24uni29: Downloading 1 format(s): mp4-1116
[download] Destination: The Course Overview [0_j24uni29].mp4
[download] 100% of 36.29MiB in 00:04
[download] Downloading video 2 of 52
[safari:api] 9781838640880/video1_2: Downloading part JSON
[safari] 9781838640880-video1_2: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_2: Downloading webpage
[Kaltura] 0_bam6t353: Downloading video info JSON
[Kaltura] 0_bam6t353: Checking mp4-532 URL
[Kaltura] 0_bam6t353: Downloading m3u8 information
[info] 0_bam6t353: Downloading 1 format(s): mp4-532
[download] Destination: Duration and Size Assertions [0_bam6t353].mp4
[download] 100% of 50.28MiB in 00:03
[download] Downloading video 3 of 52
[safari:api] 9781838640880/video1_3: Downloading part JSON
[safari] 9781838640880-video1_3: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_3: Downloading webpage
[Kaltura] 0_ryp0vswg: Downloading video info JSON
[Kaltura] 0_ryp0vswg: Checking mp4-524 URL
[Kaltura] 0_ryp0vswg: Downloading m3u8 information
[info] 0_ryp0vswg: Downloading 1 format(s): mp4-537
[download] Destination: HTML Assertions [0_ryp0vswg].mp4
[download] 100% of 15.50MiB in 00:02
[download] Downloading video 4 of 52
[safari:api] 9781838640880/video1_4: Downloading part JSON
[safari] 9781838640880-video1_4: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_4: Downloading webpage
[Kaltura] 0_a5wqpxif: Downloading video info JSON
[Kaltura] 0_a5wqpxif: Checking mp4-533 URL
[Kaltura] 0_a5wqpxif: Downloading m3u8 information
[info] 0_a5wqpxif: Downloading 1 format(s): mp4-566
[download] Destination: XML and XML Schema Assertions [0_a5wqpxif].mp4
[download] 100% of 33.48MiB in 00:03
[download] Downloading video 5 of 52
[safari:api] 9781838640880/video1_5: Downloading part JSON
[safari] 9781838640880-video1_5: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_5: Downloading webpage
[Kaltura] 0_2fuux6mp: Downloading video info JSON
[Kaltura] 0_2fuux6mp: Checking mp4-532 URL
[Kaltura] 0_2fuux6mp: Downloading m3u8 information
[info] 0_2fuux6mp: Downloading 1 format(s): mp4-594
[download] Destination: Response Assertion [0_2fuux6mp].mp4
[download] 100% of 25.71MiB in 00:03
[download] Downloading video 6 of 52
[safari:api] 9781838640880/video1_6: Downloading part JSON
[safari] 9781838640880-video1_6: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_6: Downloading webpage
[Kaltura] 0_uppk90n6: Downloading video info JSON
[Kaltura] 0_uppk90n6: Checking mp4-533 URL
[Kaltura] 0_uppk90n6: Downloading m3u8 information
[info] 0_uppk90n6: Downloading 1 format(s): mp4-657
[download] Destination: XPath and JSON Assertions [0_uppk90n6].mp4
[download] 100% of 87.99MiB in 00:07
[download] Downloading video 7 of 52
[safari:api] 9781838640880/video1_7: Downloading part JSON
[safari] 9781838640880-video1_7: Downloading kaltura session JSON
[Kaltura] 9781838640880-video1_7: Downloading webpage
[Kaltura] 0_ub2qwmkb: Downloading video info JSON
[Kaltura] 0_ub2qwmkb: Checking mp4-521 URL

ERROR: Interrupted by user
```

**Pull request closed without merging**